firestore/fromRef.ts

```diff
@@ -15,44 +15,21 @@
  * limitations under the License.
  */
 
-import firebase from 'firebase/app';
-import {Observable} from 'rxjs';
-
-type DocumentReference = firebase.firestore.DocumentReference;
-type SnapshotListenOptions = firebase.firestore.SnapshotListenOptions;
-type Query = firebase.firestore.Query;
-type DocumentSnapshot = firebase.firestore.DocumentSnapshot;
-type QuerySnapshot = firebase.firestore.QuerySnapshot;
+// TODO figure out what is wrong with the types...
+import { onSnapshot } from 'firebase/firestore';
+import { Observable } from 'rxjs';
+import { DocumentReference, DocumentData, SnapshotListenOptions, Query, DocumentSnapshot, QuerySnapshot } from './interfaces';
 
 /* eslint-disable @typescript-eslint/no-explicit-any */
-function _fromRef(
-    ref: any,
-    options: SnapshotListenOptions | undefined,
+export function fromRef<T=DocumentData>(ref: DocumentReference<T>, options?: SnapshotListenOptions): Observable<DocumentSnapshot<T>>;
+export function fromRef<T=DocumentData>(red: Query<T>, options?: SnapshotListenOptions): Observable<QuerySnapshot<T>>;
```

Comment:
caught a small typo:

```suggestion
export function fromRef<T=DocumentData>(ref: Query<T>, options?: SnapshotListenOptions): Observable<QuerySnapshot<T>>;
```

Reply:
fixed